api/src/main/java/org/openmrs/hl7/handler/ORUR01Handler.java

```diff
@@ -349,27 +349,25 @@ private Message processORU_R01(ORU_R01 oru) throws HL7Exception {
 					} else {
 						errorInHL7Queue = new HL7Exception(Context.getMessageSourceService().getMessage(
 						    "Hl7.proposed.concept.name.empty"), proposingException);
+						
 						break;//stop any further processing of current message
 					}
 					
 				}
 				catch (HL7Exception e) {
 					errorInHL7Queue = e;
-				}
-				finally {
-					// Handle obs-level exceptions
-					if (errorInHL7Queue != null) {
-						throw new HL7Exception(Context.getMessageSourceService().getMessage(
-						    "ORUR01.error.improperlyFormattedOBX",
-						    new Object[] { PipeParser.encode(obx, new EncodingCharacters('|', "^~\\&")) }, null),
-						        HL7Exception.DATA_TYPE_ERROR, errorInHL7Queue);
+					throw new HL7Exception(Context.getMessageSourceService().getMessage(
+            finally {      "ORUR01.error.improperlyFormattedOBX"));
```

Comment:
Leave it as `HL7Exception(Context.getMessageSourceService().getMessage("ORUR01.error.improperlyFormattedOBX"));` and move on to changing the finally block if the tests are passing at this stage

Reply:
Also the sonar issue is to remove the `throws` statement. Try to replace it with a more graceful way of handling the error

Reply:
I am really confused